lib/ExecutionControlEpic/ControlPanelFeature/Containers/Panel.js

```diff
@@ -1,32 +1,127 @@
 "use babel";
 // @flow
 
+import { Map } from "immutable";
 import { connect } from "react-redux";
 import { compose, withState, withProps, lifecycle } from "recompose";
-import type { State } from "../../../GlobalSystem/types.js.flow";
+import { splitControlPanel } from "../Actions/SplitControlPanel";
 import Panel from "../Presenters/Panel";
+import { getPackagePanelsFromState } from "../Model/Factory";
 import type { DisplayParams } from "../Types/types";
+import {
+  selectDiagnosticsReducer,
+  selectTasksReducer,
+  selectPlansReducer,
+} from "../../../GlobalSystem/Selectors";
+import type { State } from "../../../GlobalSystem/types.js.flow";
 import { runTask } from "../../TaskExecutionFeature/Actions/RunTask";
-import type { PlanConfig } from "../../PlanConfigurationFeature/Types/types.js.flow";
+import { killTask } from "../../TaskExecutionFeature/Actions/KillTask";
+import {
+  selectPertinentTaskID,
+  selectTasksOfPlan,
+} from "../../TaskExecutionFeature/Selectors/Tasks";
+import type { Task } from "../../TaskExecutionFeature/Types/types";
 import { unpinPlanConfig } from "../../PlanConfigurationFeature/Actions/UnpinPlanConfig";
 import { pinPlanConfig } from "../../PlanConfigurationFeature/Actions/PinPlanConfig";
 import { removePlanConfig } from "../../PlanConfigurationFeature/Actions/RemovePlanConfig";
 import { openPlanConfigurer } from "../../PlanConfigurationFeature/Actions/OpenPlanConfigurer";
-import { getPackagePanelsFromState } from "../Model/Factory";
+import { selectAllPlans } from "../../PlanConfigurationFeature/Selectors/PlanConfigs";
+import type { PlanConfig } from "../../PlanConfigurationFeature/Types/types.js.flow";
+import ExecutionsController from "../../LanguageServerProtocolFeature/Model/ExecutionsController";
 import type { Range } from "../../LanguageServerProtocolFeature/Types/standard";
-import {
-  ExecutionSelector,
-  DiagnosticsSelector,
-  TaskChangeHandler,
-  PertinentTaskSelector,
-  TaskState,
-} from "../../DiagnosticsFeature/Containers/ToolDiagnosticsTraveller";
 import { jumpTo } from "../../DiagnosticsFeature/Actions/JumpTo";
-import { killTask } from "../../TaskExecutionFeature/Actions/KillTask";
-import ExecutionsController from "../../LanguageServerProtocolFeature/Model/ExecutionsController";
-import { splitControlPanel } from "../Actions/SplitControlPanel";
+import { selectDiagnosticsOfTask } from "../../DiagnosticsFeature/Selectors/Diagnostics";
 import type { DevTool } from "../../DevToolsSummaryFeature/Types/types.js.flow";
 
+const selectPertinentTaskIDForPanel = (
+  state,
+  toolId: string,
+  plan: PlanConfig,
+) => {
+  if (plan === undefined) return "";
```

Comment:
Giving the type, plan cannot be undefined. Update the type, or check if it's really necessary to do this test.